app/api/orders.py

```diff
@@ -348,6 +356,16 @@ def before_update_object(self, order, data, view_kwargs):
             if order.order_notes and data['order_notes'] not in order.order_notes.split(","):
                 data['order_notes'] = '{},{}'.format(order.order_notes, data['order_notes'])
 
+        if data.get('payment_mode') == 'free':
```

Comment:
@shreyanshdwivedi I think this should have an additional check saying `and data.get('amount')>0`